diff -upr -x .deps -x .libs -x '*.la' atk-2.10.0.old/atk/atk.symbols atk-2.10.0/atk/atk.symbols
--- atk-2.10.0.old/atk/atk.symbols	2013-09-13 15:00:44 +0000
+++ atk-2.10.0/atk/atk.symbols	2014-05-17 16:34:24 +0000
@@ -107,6 +107,7 @@
 	atk_object_get_n_accessible_children
 	atk_object_get_name
 	atk_object_get_parent
+	atk_object_get_object_locale
 	atk_object_get_role
 	atk_object_get_type
 	atk_object_initialize
