diff -upr -x .deps -x .libs -x '*.la' -x '*.lo' gtk+-3.16.4.old/gdk/Makefile.in gtk+-3.16.4/gdk/Makefile.in
--- gtk+-3.16.4.old/gdk/Makefile.in	2015-06-10 14:48:08 +0000
+++ gtk+-3.16.4/gdk/Makefile.in	2015-07-12 16:34:22 +0000
@@ -1841,9 +1841,9 @@ dist-hook: ../build/win32/vs9/gdk.vcproj
 	$(CPP) -P - <$(top_srcdir)/build/win32/vs10/gdk.vcxproj.filtersin >$@
 	rm libgdk.vs10.sourcefiles.filters
 
-install-data-local: install-ms-lib install-def-file
+install-data-local: install-ms-lib
 
-uninstall-local: uninstall-ms-lib uninstall-def-file
+uninstall-local: uninstall-ms-lib
 	rm -f $(DESTDIR)$(configexecincludedir)/gdkconfig.h
 
 # if srcdir!=builddir, clean out maintainer-clean files from builddir
diff -upr -x .deps -x .libs -x '*.la' -x '*.lo' gtk+-3.16.4.old/gtk/deprecated/gtkaction.c gtk+-3.16.4/gtk/deprecated/gtkaction.c
--- gtk+-3.16.4.old/gtk/deprecated/gtkaction.c	2015-03-02 19:34:03 +0000
+++ gtk+-3.16.4/gtk/deprecated/gtkaction.c	2015-07-12 16:34:22 +0000
@@ -518,7 +518,7 @@ gtk_action_class_init (GtkActionClass *k
                                    g_param_spec_boolean ("always-show-image",
                                                          P_("Always show image"),
                                                          P_("Whether the image will always be shown"),
-                                                         FALSE,
+                                                         TRUE,
                                                          GTK_PARAM_READWRITE | G_PARAM_CONSTRUCT));
 
   /**
diff -upr -x .deps -x .libs -x '*.la' -x '*.lo' gtk+-3.16.4.old/gtk/deprecated/gtkimagemenuitem.c gtk+-3.16.4/gtk/deprecated/gtkimagemenuitem.c
--- gtk+-3.16.4.old/gtk/deprecated/gtkimagemenuitem.c	2015-05-16 00:48:48 +0000
+++ gtk+-3.16.4/gtk/deprecated/gtkimagemenuitem.c	2015-07-12 16:34:22 +0000
@@ -232,7 +232,7 @@ gtk_image_menu_item_class_init (GtkImage
                                    g_param_spec_boolean ("always-show-image",
                                                          P_("Always show image"),
                                                          P_("Whether the image will always be shown"),
-                                                         FALSE,
+                                                         TRUE,
                                                          GTK_PARAM_READWRITE | G_PARAM_CONSTRUCT | G_PARAM_DEPRECATED));
 
   /**
diff -upr -x .deps -x .libs -x '*.la' -x '*.lo' gtk+-3.16.4.old/gtk/gtkbutton.c gtk+-3.16.4/gtk/gtkbutton.c
--- gtk+-3.16.4.old/gtk/gtkbutton.c	2015-05-16 00:48:48 +0000
+++ gtk+-3.16.4/gtk/gtkbutton.c	2015-07-12 16:34:22 +0000
@@ -365,7 +365,7 @@ gtk_button_class_init (GtkButtonClass *k
      g_param_spec_boolean ("always-show-image",
                            P_("Always show image"),
                            P_("Whether the image will always be shown"),
-                           FALSE,
+                           TRUE,
                            GTK_PARAM_READWRITE|G_PARAM_CONSTRUCT|G_PARAM_EXPLICIT_NOTIFY);
 
   g_object_class_install_properties (gobject_class, LAST_PROP, props);
diff -upr -x .deps -x .libs -x '*.la' -x '*.lo' gtk+-3.16.4.old/gtk/gtkscrolledwindow.c gtk+-3.16.4/gtk/gtkscrolledwindow.c
--- gtk+-3.16.4.old/gtk/gtkscrolledwindow.c	2015-06-10 13:49:02 +0000
+++ gtk+-3.16.4/gtk/gtkscrolledwindow.c	2015-07-12 16:34:45 +0000
@@ -596,7 +596,7 @@ gtk_scrolled_window_class_init (GtkScrol
                                    g_param_spec_boolean ("overlay-scrolling",
                                                          P_("Overlay Scrolling"),
                                                          P_("Overlay scrolling mode"),
-                                                         TRUE,
+                                                         FALSE,
                                                          GTK_PARAM_READWRITE|G_PARAM_EXPLICIT_NOTIFY));
 
   /**
@@ -4052,8 +4052,8 @@ gtk_scrolled_window_update_use_indicator
 
   use_indicators = priv->overlay_scrolling;
 
-  if (g_strcmp0 (g_getenv ("GTK_OVERLAY_SCROLLING"), "0") == 0)
-    use_indicators = FALSE;
+  if (g_strcmp0 (g_getenv ("GTK_OVERLAY_SCROLLING"), "1") == 0)
+    use_indicators = TRUE;
 
   if (priv->use_indicators != use_indicators)
     {
