diff -upr jasper-1.900.1/src/appl/tmrdemo.c jasper-1.900.1.new/src/appl/tmrdemo.c
--- jasper-1.900.1/src/appl/tmrdemo.c	2007-01-19 21:43:08 +0000
+++ jasper-1.900.1.new/src/appl/tmrdemo.c	2013-12-29 22:42:52 +0000
@@ -43,7 +43,7 @@ int main(int argc, char **argv)
 	printf("zero time %.3f us\n", t * 1e6);
 
 	jas_tmr_start(&tmr);
-	sleep(1);
+	//sleep(1);
 	jas_tmr_stop(&tmr);
 	t = jas_tmr_get(&tmr);
 	printf("time delay %.8f s\n", t);
