diff -upr libtasn1-3.4.old/config.h libtasn1-3.4/config.h
--- libtasn1-3.4.old/config.h	2014-03-24 22:51:20 +0000
+++ libtasn1-3.4/config.h	2014-03-24 22:53:52 +0000
@@ -558,7 +558,7 @@
 #define HAVE_WINSOCK2_H 1
 
 /* Define to 1 if you have the `_ftelli64' function. */
-#define HAVE__FTELLI64 1
+#undef HAVE__FTELLI64
 
 /* Define to 1 if you have the `_ftime' function. */
 #define HAVE__FTIME 1
diff -upr libtasn1-3.4.old/gl/msvc-inval.c libtasn1-3.4/gl/msvc-inval.c
--- libtasn1-3.4.old/gl/msvc-inval.c	2014-03-24 22:52:55 +0000
+++ libtasn1-3.4/gl/msvc-inval.c	2014-03-24 22:50:23 +0000
@@ -28,7 +28,7 @@
 
 # if MSVC_INVALID_PARAMETER_HANDLING == DEFAULT_HANDLING
 
-static void cdecl
+static void __cdecl
 gl_msvc_invalid_parameter_handler (const wchar_t *expression,
                                    const wchar_t *function,
                                    const wchar_t *file,
@@ -45,7 +45,7 @@ gl_msvc_invalid_parameter_handler (const
 
 #  if defined _MSC_VER
 
-static void cdecl
+static void __cdecl
 gl_msvc_invalid_parameter_handler (const wchar_t *expression,
                                    const wchar_t *function,
                                    const wchar_t *file,
@@ -94,7 +94,7 @@ gl_msvc_inval_current (void)
     }
 }
 
-static void cdecl
+static void __cdecl
 gl_msvc_invalid_parameter_handler (const wchar_t *expression,
                                    const wchar_t *function,
                                    const wchar_t *file,
