diff -upr -x .deps -x .libs -x '*.la' -x '*.lo' nettle-3.1/config.h.in nettle-3.1.new/config.h.in
--- nettle-3.1/config.h.in	2015-04-07 19:00:08 +0000
+++ nettle-3.1.new/config.h.in	2015-07-05 18:55:52 +0000
@@ -142,9 +142,6 @@
 /* The size of `short', as computed by sizeof. */
 #undef SIZEOF_SHORT
 
-/* The size of `size_t', as computed by sizeof. */
-#undef SIZEOF_SIZE_T
-
 /* The size of `void*', as computed by sizeof. */
 #undef SIZEOF_VOIDP
 
diff -upr -x .deps -x .libs -x '*.la' -x '*.lo' nettle-3.1/configure nettle-3.1.new/configure
--- nettle-3.1/configure	2015-04-07 19:00:08 +0000
+++ nettle-3.1.new/configure	2015-07-05 18:54:38 +0000
@@ -5444,39 +5444,7 @@ cat >>confdefs.h <<_ACEOF
 _ACEOF
 
 
-# The cast to long int works around a bug in the HP C Compiler
-# version HP92453-01 B.11.11.23709.GP, which incorrectly rejects
-# declarations like `int a3[[(sizeof (unsigned char)) >= 0]];'.
-# This bug is HP SR number 8606223364.
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking size of size_t" >&5
-$as_echo_n "checking size of size_t... " >&6; }
-if ${ac_cv_sizeof_size_t+:} false; then :
-  $as_echo_n "(cached) " >&6
-else
-  if ac_fn_c_compute_int "$LINENO" "(long int) (sizeof (size_t))" "ac_cv_sizeof_size_t"        "$ac_includes_default"; then :
-
-else
-  if test "$ac_cv_type_size_t" = yes; then
-     { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error 77 "cannot compute sizeof (size_t)
-See \`config.log' for more details" "$LINENO" 5; }
-   else
-     ac_cv_sizeof_size_t=0
-   fi
-fi
-
-fi
-{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_sizeof_size_t" >&5
-$as_echo "$ac_cv_sizeof_size_t" >&6; }
-
-
-
-cat >>confdefs.h <<_ACEOF
-#define SIZEOF_SIZE_T $ac_cv_sizeof_size_t
-_ACEOF
-
-
+#AC_CHECK_SIZEOF(size_t)
 
 for ac_header in openssl/blowfish.h openssl/des.h openssl/cast.h openssl/aes.h openssl/ecdsa.h
 do :
diff -upr -x .deps -x .libs -x '*.la' -x '*.lo' nettle-3.1/configure.ac nettle-3.1.new/configure.ac
--- nettle-3.1/configure.ac	2015-04-07 19:00:08 +0000
+++ nettle-3.1.new/configure.ac	2015-07-05 18:53:40 +0000
@@ -183,7 +183,7 @@ AC_TYPE_UID_T
 AC_TYPE_SIZE_T
 AC_HEADER_TIME
 AC_CHECK_SIZEOF(long)
-AC_CHECK_SIZEOF(size_t)
+#AC_CHECK_SIZEOF(size_t)
 
 AC_CHECK_HEADERS([openssl/blowfish.h openssl/des.h openssl/cast.h openssl/aes.h openssl/ecdsa.h],,
 [enable_openssl=no
