diff -upr -x .deps -x .libs -x '*.la' -x '*.lo' p11-kit-ec9e2450ba/p11-kit/Makefile.am p11-kit-ec9e2450ba.new/p11-kit/Makefile.am
--- p11-kit-ec9e2450ba/p11-kit/Makefile.am	2015-02-20 20:28:59 +0000
+++ p11-kit-ec9e2450ba.new/p11-kit/Makefile.am	2015-07-06 16:01:39 +0000
@@ -73,6 +73,7 @@ libp11_kit_testable_la_CFLAGS = \
 	$(LIBFFI_CFLAGS) \
 	$(NULL)
 
+libp11_kit_la_LIBADD += -lws2_32
 else
 
 libp11_kit_testable_la_CFLAGS = \
